subprojects/gradle/kubernetes/src/test/kotlin/com/avito/utils/gradle/KubernetesRequestMetadataProviderTest.kt

```diff
@@ -0,0 +1,49 @@
+package com.avito.utils.gradle
+
+import com.avito.http.internal.RequestMetadataProvider
+import com.avito.truth.ResultSubject
+import com.google.common.truth.Truth.assertThat
+import okhttp3.MediaType
+import okhttp3.Request
+import okhttp3.RequestBody
+import org.junit.jupiter.api.Test
+
+internal class KubernetesRequestMetadataProviderTest {
+
+    @Test
+    fun `metadata contains get deployments request`() {
+        val request = Request.Builder()
+            .url("https://k8s.service:8443/apis/apps/v1/namespaces/android-emulator/deployments")
```

Comment:
Can we create a `KubernetesClient` here to simulate a request?
It may be easier to map what we use in logic to metrics.

Reply:
i don't know how to extract formed request from KubernetesClient add pass it here

Reply:
I mean to intercept a real request from KubernetesClient against a mock server.

Reply:
This one: https://github.com/fabric8io/kubernetes-client/tree/master/kubernetes-server-mock

Reply:
Yes, it would be nice improvement of this tests, but i decided to postpone it